stag-library-compiler/src/main/java/com/vimeo/stag/processor/generators/TypeAdapterGenerator.java

```diff
@@ -78,11 +79,10 @@ public TypeSpec getTypeAdapterSpec() {
                 .addModifiers(Modifier.PUBLIC)
                 .addParameter(Gson.class, "gson");
 
-        TypeSpec.Builder adapterBuilder =
-                TypeSpec.classBuilder(mInfo.getTypeAdapterClassName())
-                        .addModifiers(Modifier.PUBLIC, Modifier.FINAL)
-                        .superclass(ParameterizedTypeName.get(ClassName.get(TypeAdapter.class),
-                                                              typeVariableName));
+        String className = FileGenUtils.desanitizeCode(mInfo.getTypeAdapterClassName());
```

Comment:
The gist of it is that whenever we are passing a string into a `CodeBlock` object, it is parsed and `$` are interpreted as formatting characters and it breaks things. The solution is to pass in `$$` so that it replaces it with `$`. The problem is that other classes take the String literal at face value and `$$` is interpreted as `$$` which causes problems, so we need to "desanitize" it.

let's talk irl about it